app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -417,7 +417,14 @@ class HomeFragment : Fragment() {
             frecencyConfig = FrecencyThresholdOption.SKIP_ONE_TIME_PAGES,
             providerConfig = TopSitesProviderConfig(
                 showProviderTopSites = settings.showContileFeature,
-                maxThreshold = TOP_SITES_PROVIDER_MAX_THRESHOLD
+                maxThreshold = TOP_SITES_PROVIDER_MAX_THRESHOLD,
+                providerFilter = { topSite ->
+                    when (store.state.search.selectedOrDefaultSearchEngine?.name) {
+                        "Amazon.com" -> topSite.title != "Amazon"
```

Comment:
nit: since these two topsites' strings are used in separate places, would it make sense to wrap these in a constant?